targets/TARGET_STM/qspi_api.c

```diff
@@ -128,11 +266,94 @@ void qspi_prepare_command(const qspi_command_t *command, QSPI_CommandTypeDef *st
     }
 
     st_command->NbData = 0;
+    debug_if(qspi_api_c_debug, "qspi_prepare_command Out: InstructionMode %x Instruction %x AddressMode %x AddressSize %x Address %x DataMode %x\n",
+        st_command->InstructionMode, st_command->Instruction, st_command->AddressMode, st_command->AddressSize, st_command->Address, st_command->DataMode);
 }
+#endif /* OCTOSPI */
+
+
+#if defined(OCTOSPI1)
+qspi_status_t qspi_init(qspi_t *obj, PinName io0, PinName io1, PinName io2, PinName io3, PinName sclk, PinName ssel, uint32_t hz, uint8_t mode)
+{
+    OSPIM_CfgTypeDef OSPIM_Cfg_Struct = {0};
+    debug_if(qspi_api_c_debug, "qspi_init mode %u\n", mode);
 
+    __HAL_RCC_OSPI2_CLK_ENABLE();
+    __HAL_RCC_OSPIM_CLK_ENABLE();
 
+    __HAL_RCC_OSPI2_FORCE_RESET();
+    __HAL_RCC_OSPI2_RELEASE_RESET();
+
+
+    // Reset handle internal state
+    obj->handle.State = HAL_OSPI_STATE_RESET;
+
+    // Set default OCTOSPI handle values
+    obj->handle.Init.DualQuad = HAL_OSPI_DUALQUAD_DISABLE;
+    obj->handle.Init.MemoryType = HAL_OSPI_MEMTYPE_MICRON;
+    obj->handle.Init.ClockPrescaler = 4;
+    obj->handle.Init.FifoThreshold = 4;
+    obj->handle.Init.SampleShifting = HAL_OSPI_SAMPLE_SHIFTING_NONE;
+    obj->handle.Init.DeviceSize = POSITION_VAL(QSPI_FLASH_SIZE_DEFAULT) - 1;
+    obj->handle.Init.ChipSelectHighTime = 3;
+    obj->handle.Init.FreeRunningClock = HAL_OSPI_FREERUNCLK_DISABLE;
+    obj->handle.Init.WrapSize = HAL_OSPI_WRAP_NOT_SUPPORTED;
+    obj->handle.Init.ClockMode = mode == 0 ? HAL_OSPI_CLOCK_MODE_0 : HAL_OSPI_CLOCK_MODE_3;
+    obj->handle.Init.DelayHoldQuarterCycle = HAL_OSPI_DHQC_ENABLE;
+    obj->handle.Init.ChipSelectBoundary = 0;
+
+    QSPIName qspiio0name = (QSPIName)pinmap_peripheral(io0, PinMap_QSPI_DATA0);
+    QSPIName qspiio1name = (QSPIName)pinmap_peripheral(io1, PinMap_QSPI_DATA1);
+    QSPIName qspiio2name = (QSPIName)pinmap_peripheral(io2, PinMap_QSPI_DATA2);
+    QSPIName qspiio3name = (QSPIName)pinmap_peripheral(io3, PinMap_QSPI_DATA3);
+    QSPIName qspiclkname = (QSPIName)pinmap_peripheral(sclk, PinMap_QSPI_SCLK);
+    QSPIName qspisselname = (QSPIName)pinmap_peripheral(ssel, PinMap_QSPI_SSEL);
+
+    QSPIName qspi_data_first = (QSPIName)pinmap_merge(qspiio0name, qspiio1name);
+    QSPIName qspi_data_second = (QSPIName)pinmap_merge(qspiio2name, qspiio3name);
+    QSPIName qspi_data_third = (QSPIName)pinmap_merge(qspiclkname, qspisselname);
+
+    if (qspi_data_first != qspi_data_second || qspi_data_second != qspi_data_third ||
+            qspi_data_first != qspi_data_third) {
+        debug_if(qspi_api_c_debug, "QSPI_STATUS_INVALID_PARAMETER error\n");
+        return QSPI_STATUS_INVALID_PARAMETER;
+    }
+
+    // tested all combinations, take first
+    obj->handle.Instance = ((OCTOSPI_TypeDef *) OCTOSPI2_R_BASE);
```

Comment:
Same comment, this value was hardcoded to OSPI2 for test...

Reply:
done